src/main/java/de/zalando/aruha/nakadi/repository/zookeeper/ZookeeperUtils.java

```diff
@@ -0,0 +1,28 @@
+package de.zalando.aruha.nakadi.repository.zookeeper;
+
+import org.apache.curator.framework.recipes.locks.InterProcessLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.Callable;
+
+public class ZookeeperUtils {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ZookeeperUtils.class);
+
+    private ZookeeperUtils() {
+    }
+
+    public static <V> V runLocked(final Callable<V> callable, final InterProcessLock lock) throws Exception {
+        try {
+            lock.acquire();
+            return callable.call();
```

Comment:
Well, in general there is an approach for that: if method returned with exception, than it's up to this method to rollback all the data back. So in this case it's like that.
Even more - if acqure() throwed Exception, than it's guaranteed that release() will throw exception :)